components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -55,55 +115,269 @@ set_updated_group (const FieldGroup<Real>& group)
+          dz("dz",m_num_cols,nlev_packs),
+          zt_grid("zt_grid",m_num_cols,nlev_packs),
+          zi_grid("zi_grid",m_num_cols,nlevi_packs),
+          wtracer_sfc("wtracer_sfc",m_num_cols,num_tracer_packs),
+          wm_zt("wm_zt",m_num_cols,nlev_packs),
+          exner("exner",m_num_cols,nlev_packs),
+          thlm("thlm",m_num_cols,nlev_packs),
+          qw("qw",m_num_cols,nlev_packs),
+          cloud_frac("cloud_frac",m_num_cols,nlev_packs);
+
+  view_3d tracers("tracers",m_num_cols,m_num_levs,num_tracer_packs);
+
+  shoc_preamble.set_variables(m_num_cols,m_num_levs,m_num_tracers,nlev_packs,num_tracer_packs,t,alst,
+                              zi,zm,pmid,pdel,omega,shf,cflx_k0,wsx,wsy,shoc_qv,Q,shoc_ql,tke,
+                              s,u,v,
+                              rrho,rrho_i,thv,dz,zt_grid,zi_grid,wpthlp_sfc,wprtp_sfc,upwp_sfc,vpwp_sfc,
+                              wtracer_sfc,wm_zt,exner,thlm,qw,cloud_frac,tracers);
 
-  auto q_ptr = Q->get_view<Host>().data();
+  // Input Variables:
+  input.host_dx     = m_shoc_fields_in["host_dx"].get_reshaped_view<const Pack1d*>();
+  input.host_dy     = m_shoc_fields_in["host_dy"].get_reshaped_view<const Pack1d*>();
+  input.zt_grid     = shoc_preamble.zt_grid;
+  input.zi_grid     = shoc_preamble.zi_grid;
+  input.pres        = pmid;
+  input.presi       = m_shoc_fields_in["pint"].get_reshaped_view<const Spack**>();
+  input.pdel        = pdel;
+  input.thv         = shoc_preamble.thv;
+  input.w_field     = shoc_preamble.wm_zt;
+  input.wthl_sfc    = shoc_preamble.wpthlp_sfc;
+  input.wqw_sfc     = shoc_preamble.wprtp_sfc;
+  input.uw_sfc      = shoc_preamble.upwp_sfc;
+  input.vw_sfc      = shoc_preamble.vpwp_sfc;
+  input.wtracer_sfc = shoc_preamble.wtracer_sfc;
+  input.exner       = shoc_preamble.exner;
+  input.phis        = m_shoc_fields_in["phis"].get_reshaped_view<const Pack1d*>();
 
-  shoc_init_f90 (q_ptr);
+  // Input/Output Variables
+  input_output.host_dse     = shoc_preamble.shoc_s;
+  input_output.tke          = shoc_preamble.tke_zt;
+  input_output.thetal       = shoc_preamble.thlm;
+  input_output.qw           = shoc_preamble.qw;
+  input_output.u_wind       = shoc_preamble.um;
+  input_output.v_wind       = shoc_preamble.vm;
+  input_output.wthv_sec     = m_shoc_fields_out["wthv_sec"].get_reshaped_view<Spack**>();
+  input_output.qtracers     = shoc_preamble.tracers;
+  input_output.tk           = m_shoc_fields_out["tk"].get_reshaped_view<Spack**>();
+  input_output.tkh          = m_shoc_fields_out["tkh"].get_reshaped_view<Spack**>();
+  input_output.shoc_cldfrac = shoc_preamble.cloud_frac;
+  input_output.shoc_ql      = shoc_preamble.shoc_ql;
 
-  Q->sync_to_dev();
+  // Output Variables
+  output.pblh     = m_shoc_fields_out["pblh"].get_reshaped_view<Pack1d*>();
+
+  view_2d shoc_ql2("shoc_ql2",m_num_cols,nlev_packs);
+  output.shoc_ql2 = shoc_ql2;
+
+  // Ouput (diagnostic)
+  view_2d shoc_mix("shoc_mix",m_num_cols,nlev_packs);
```

Comment:
Will all of these views ultimately disappear? This is time-stepping code, right, not one-time init? If so, we want to use the workspace manager for views like this so we don't bog down in view allocations and deallocations.

Reply:
With these I'm not sure since they are output variables of shoc_main. It would depend if they are needed for any post-shoc_main computation below in `run_impl()`.

Reply:
Edit: I see Luca's comment below. Yes, they will. I will add a TODO.